src/app/app.authorization.js

```diff
@@ -0,0 +1,18 @@
+angular.module('app-bootstrap').run([
+  '$rootScope', '$state', 'authenticationService', '$http', 'localStorageService',
+  function ($rootScope, $state, authenticationService, $http, localStorageService) {
+
+    $http.defaults.headers.common.AUTHORIZATION = localStorageService.get('session_token');
+
+    $rootScope.$on('$stateChangeStart', function (event, toState) {
+
+      if (!authenticationService.isLoggedIn() && toState.data.requireLogin) {
+        event.preventDefault();
```

Comment:
As the two conditions are not mutually exclusive and we only want to prevent default in those two cases, writing it outside would not be the desired behavior (actually, the page doesn't work if it's in line 8)